### PR TITLE
core[patch], openai[patch]: Stream logprobs

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/partners/openai/langchain_openai/chat_models/base.py [hunk 1]
```diff
@@ -404,12 +404,15 @@ def _stream(
             chunk = _convert_delta_to_message_chunk(
                 choice["delta"], default_chunk_class
             )
-            finish_reason = choice.get("finish_reason")
-            generation_info = (
-                dict(finish_reason=finish_reason) if finish_reason is not None else None
-            )
+            generation_info = {}
```
Comment (eyurtsev) [inline libs/partners/openai/langchain_openai/chat_models/base.py:407]: Any unit tests that we could add?

**Review by eyurtsev — APPROVED** [review]
Looks good to me.

**Comment by eyurtsev** [comment]
@baskaryan maybe update subject line that this contains finish reason?

File: libs/core/langchain_core/utils/merge.py
Comment (efriis) [inline libs/core/langchain_core/utils/merge.py:1]: Looks like this should only be used internal in core, so maybe worth `_merge.py`

**Review by efriis — APPROVED** [review]

File: libs/core/langchain_core/utils/merge.py
Comment (baskaryan) [reply libs/core/langchain_core/utils/merge.py:1]: maybe we should rename all of utils -> _utils

File: libs/core/langchain_core/utils/_merge.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            raise TypeError(
+                f'additional_kwargs["{k}"] already exists in this message,'
+                " but with a different type."
+            )
+        elif isinstance(merged[k], str):
+            merged[k] += v
+        elif isinstance(merged[k], dict):
+            merged[k] = merge_dicts(merged[k], v, merge_lists=merge_lists)
+        elif isinstance(merged[k], list):
+            if merge_lists:
```
Comment (baskaryan) [inline libs/core/langchain_core/utils/_merge.py:47]: cc @nfcampos added this param

**Review by nfcampos — APPROVED** [review]

**Pull request merged**